csr/csr.go

```diff
@@ -31,12 +31,12 @@ var badSignatureAlgorithms = map[x509.SignatureAlgorithm]bool{
 }
 
 var (
-	invalidPubKey       = errors.New("invalid public key in CSR")
-	unsupportedSigAlg   = errors.New("signature algorithm not supported")
-	invalidSig          = errors.New("invalid signature on CSR")
-	invalidEmailPresent = errors.New("CSR contains one or more email address fields")
-	invalidIPPresent    = errors.New("CSR contains one or more IP address fields")
-	invalidNoDNS        = errors.New("at least one DNS name is required")
+	invalidPubKey       = berrors.MalformedError("invalid public key in CSR")
```

Comment:
This is a good example of the question I had on the error policy. I think generally we should lean on the side of using stdlib errors unless we are trying to express something specific with the berrors. For instance, here everything is the same berror type. We could just as well return `errors.New` from all of these, and have the caller be responsible for turning that into a user-facing MalformedProblem.

If this function could return a variety of types, which would imply a variety of ProblemDetails, using berrors would make sense.